email_verification/models.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Models for email verification
+"""
+
+# Imports #####################################################################
+
+from datetime import datetime, timedelta, timezone
+from urllib.parse import urljoin
+
+from django.conf import settings
+from django.core.mail import send_mail
+from django.core.urlresolvers import reverse
+from django.db import models
+from django.utils.crypto import get_random_string
+
+
+# Models ######################################################################
+
+class EmailVerification(models.Model):
```

Comment:
@antoviaque I evaluated [django-simple-email-confirmation](https://github.com/mfogel/django-simple-email-confirmation), [django-registration](https://github.com/macropin/django-registration), and [django-userena](https://github.com/bread-and-pepper/django-userena) for this. All of them implement email verification but tie email addresses to the `User` model. In this case we also want to verify an email address that belongs to the `BetaTestApplication` model. I considered using django-simple-email-confirmation and adding some mappings between its `EmailAddress` model and out `BetaTestApplication` model, but in the end I decided it would be cleaner and easier to just implement the basic email verification functionality as a separate app.